src/components/Tax.tsx

```diff
@@ -1,21 +1,24 @@
-import React from 'react';
-import PropTypes from 'prop-types';
-import { StyleSheet, Text, View, TextInput } from 'react-native';
+import React, { ForwardedRef, forwardRef } from 'react';
+import { Text, View, TextInput } from 'react-native';
 
-export default function Tax({ tax, setTax }) {
+type TaxProps = {
+  tax: number,
+  setTax: (tax: number) => void,
+};
+
+export default forwardRef(({ tax, setTax }: TaxProps, ref: ForwardedRef<TextInput>) => {
```

Comment:
I did a forward ref so that when a user successfully submits a transaction, the fields are reset. However, since this field is using `defaultText` and not the `value` prop -- what this means is that the user's input is not being cleared when the app state is reset.

I did try using `value` except I couldn't type decimals in the input so as a workaround (and stackoverflow) opted to use a ref to then clear the field. 

As a result the changes for this component involve:
* Removing proptypes and defining the type in `TaxProps` 
* Adding a ref to `TextInput`

Reply:
before the ref changes note how tax input doesn't change (though the tax summary goes back to 0.00) 

https://github.com/tally-team/tally-frontend/assets/24977851/a49a7423-07d1-413d-8f55-03cb81ea51f1

Reply:
> using defaultText and not the value prop -- what this means is that the user's input is not being cleared when the app state is reset.

Is it because by setting a default value, the user's input will become the new "default value"? so the default value or/and user input is never cleared?